From 6854eb7d239398daab650d34019858f6364e9066 Mon Sep 17 00:00:00 2001
From: Naveen Cherukuri <naveen.cherukuri@xilinx.com>
Date: Tue, 4 Dec 2018 16:15:22 +0530
Subject: [PATCH] kmssink: Avoiding get_property to take ownership of object
 members

- Double free will happen if application frees string retuned by _get_property

Upstream-Status: Pending
Signed-off-by: Naveen Cherukuri <naveen.cherukuri@xilinx.com>
---
 sys/kms/gstkmssink.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sys/kms/gstkmssink.c b/sys/kms/gstkmssink.c
index 4a74baf..70c1f20 100644
--- a/sys/kms/gstkmssink.c
+++ b/sys/kms/gstkmssink.c
@@ -1946,10 +1946,10 @@ gst_kms_sink_get_property (GObject * object, guint prop_id,
 
   switch (prop_id) {
     case PROP_DRIVER_NAME:
-      g_value_take_string (value, sink->devname);
+      g_value_set_string (value, sink->devname);
       break;
     case PROP_BUS_ID:
-      g_value_take_string (value, sink->bus_id);
+      g_value_set_string (value, sink->bus_id);
       break;
     case PROP_CONNECTOR_ID:
       g_value_set_int (value, sink->conn_id);
-- 
2.7.4

